components/ws-manager-mk2/service/manager.go

```diff
@@ -1494,3 +1495,21 @@ func (wav *workspaceActivityVec) getWorkspaceActivityCounts() (active, notActive
 
 	return
 }
+
+func getLastActivity(ws *workspacev1.Workspace) *time.Time {
+	lastActivity := ws.Status.LastActivity
+	if lastActivity != nil {
+		return &lastActivity.Time
+	}
+
+	// In case we don't have a record of the workspace's last activity, check for the FirstUserActivity condition
+	// to see if the lastActivity got lost on a manager restart.
+	if ws.IsConditionTrue(workspacev1.WorkspaceConditionFirstUserActivity) {
+		now := time.Now().UTC()
+		lastActivityStatus := metav1.NewTime(now)
+		return &lastActivityStatus.Time
+	}
```

Comment:
Is this still needed when the lastActivity is stored inside the CRD and doesn't get lost on restart?

Reply:
Removed.